supertokens/supertokens-node
# Added tests

## Related Issues
- #32 

## Changes
- Added two process states for testing
- Process state checks in querier.ts and sessionRecipe.ts
- fixed typo in utils.ts
- Tests added:
   - Config tests:
      - test various inputs for appInfo 
      - test using zero, one and two recipe modules 
      - test config for session module 
   - Handshake test
     - test that once the info is loaded, it doesn't query again 
   - Middleware test
     - check that disabling default API actually disables it (for session) 
   - Querier test
     - Test that if the querier throws an error from a recipe, that recipe's ID is there 
     - Check that once the API version is there, it doesn't need to query again 
     - Check that rid is added to the header iff it's a "/recipe" || "/recipe/*" request. 
   - RecipeModuleManager tests
     - Check that querier has been inited when we call supertokens.init 
     - Check that modules have been inited when we call supertokens.init
     - Test various inputs to routing (if it accepts or not) 
     -  Test various inputs to errorHandler (if it accepts or not) 
     - If an error handler in a recipe throws an error, that error next to go to the user's error handler (done)
     - Error thrown from APIs implemented by recipes must not go unhandled (done)
     - Disable a default route, and then implement your own API and check that that gets called(done)
     - Test getAllCORSHeaders
   - Session tests
     - check if output headers and set cookies for create session is fine 
     - check if output headers and set cookies for refresh session is fine 
     - check if input cookies are missing, an appropriate error is thrown
   - SessionExpress tests
     -  check if disableDefaultImplementation is true, the default refresh API does not work - you get a 404

## Migration:
  - none
